### ISSUE TITLE
Is it possible to make onnx model support?

### ISSUE THREAD

**Issue by orkars** [yl4579/StyleTTS2#117]
This is coolest model that I saw. Thank you for this prefect work. For using this to different platforms, is there any supports or information to make inference?

**Comment by 81549361** [comment]
I also want to know if it can support onnx model

**Comment by yl4579** [comment]
You may refer to #39 if you just want portability. I'm not familiar with Onnx so it probably needs to be be done by someone more familiar with this.

**Comment by adrianlyjak** [comment]
Someone seems to have done this. I'm not sure what hexgrad is, whether its a researcher/hobbyist or business. 

https://huggingface.co/hexgrad/styletts2

https://hexgrad.com/

**Comment by zolero** [comment]
I'm working on this as well, I'll keep you guys posted.  You could increate interference / generate speed by quantifying the models and loading in another backend architecture like tensorrt

**Comment by logikstate** [comment]
So who is the mysterious hexgrad? and do they plan to share their most excellent onnx implementation with the rest of us mere mortals? :)

**Comment by nduartech** [comment]
> I'm working on this as well, I'll keep you guys posted. You could increate interference / generate speed by quantifying the models and loading in another backend architecture like tensorrt

Any luck yet?

**Comment by zolero** [comment]
> > I'm working on this as well, I'll keep you guys posted. You could increate interference / generate speed by quantifying the models and loading in another backend architecture like tensorrt
> 
> Any luck yet?

Add me on Discord: 450237433980780564 / Zolero

**Comment by smallbraineng** [comment]
> > > I'm working on this as well, I'll keep you guys posted. You could increate interference / generate speed by quantifying the models and loading in another backend architecture like tensorrt
> > 
> > 
> > Any luck yet?
> 
> Add me on Discord: 450237433980780564 / Zolero

following up here as well–would love to know how this is going / if I can help

**Comment by nanaghartey** [comment]
> hexgrad

the author (hexgrad) has released https://huggingface.co/hexgrad/Kokoro-82M based on StyleTTS2 architecture... Output sounds better... and there's an onnx support here https://github.com/thewh1teagle/kokoro-onnx

Only bummer is currently you can't fine tune your own models

**Comment by adrianlyjak** [comment]
Yeah, you can run the onnx models in js in your browser! https://github.com/hexgrad/kokoro/tree/main/kokoro.js
It was a bit slow when I tested (~20s for a single sentence, running in a new m3 MacBook Pro)
Xenova at HuggingFace (maintainer of transformersjs) says web gpu support coming soon: https://huggingface.co/posts/Xenova/503648859052804

**Comment by thewh1teagle** [comment]
I'm working on new styleTTS2 variant with easier training and more organized code, onnx export etc.
If someone want to join to the effort feel free to create new issue

https://github.com/thewh1teagle/styletts2-fork

**Comment by ankandrew** [comment]
> I'm working on new styleTTS2 variant with easier training and more organized code, onnx export etc. If someone want to join to the effort feel free to create new issue
> 
> https://github.com/thewh1teagle/styletts2-fork

Sounds great. I might be able to help. Is the repo currently private?

**Comment by Amarnath1906** [comment]
Anyone was able to figure out converting diffusion module to onnx?
